packages/data-point/lib/utils/index.js

```diff
@@ -89,6 +89,28 @@ function inspectProperties (obj, props, indent = '') {
 
 module.exports.inspectProperties = inspectProperties
 
+/**
+ * @param {reducer} reducer
+ * @returns {boolean}
+ */
+function reducerIsEmpty (reducer) {
+  if (!reducer) {
+    return true
+  }
+
+  if (reducer.type === 'ReducerList') {
```

Comment:
here is a concern I have, if we plan to open the API to let other devs create their custom reducers, checks like this one are the ones im worried about. and its where I go back to preferring to have reducers be casted to arrays when they are not. 

do you see that issue also? or am I not seeing something?

Reply:
just an idea a simple way to improve this would be to normalize the api so a reducer has to have a `.isEmpty()` method, this would allow each reducer to be able to check itself, and we could have this method here be more generic ?

Reply:
I don't think we should add checks for empty reducers in the first place, I only added this to make it backwards compatible. When reducers are empty, we currently just return the accumulator unchanged, but that behavior always catches me off guard:

https://github.com/ViacomInc/data-point/blob/master/packages/data-point/lib/transform-expression/resolve.js#L16

I've been planning to suggest changing that behavior, and if we do change it, we wouldn't need this `reducerIsEmpty` function anymore.  For example, if there's an empty ReducerObject, I think it should just return an empty object. If you have an empty reducer, it's probably a mistake in the first place, so we shouldn't cover up that mistake by resolving the reducer in a special way.